klayout_package/python/kqcircuits/simulations/export/simulation_validate.py

```diff
@@ -0,0 +1,119 @@
+        Args:
+            simulations: A Simulation object.
+            solutions: A Solution object.
+        Raises:
+            Errors when validation criteria are not met.
+        """
+        port_names = get_port_names(simulation)
+        sim_name = simulation.name
+        if not port_names and type(solution) in [
+            AnsysHfssSolution,
+            AnsysVoltageSolution,
+            AnsysCurrentSolution,
+            ElmerVectorHelmholtzSolution,
+            ElmerCapacitanceSolution,
+        ]:
+            raise ValidateSimError(
+                f"Simulation '{sim_name}' has no ports assigned. This is incompatible with {type(solution)}",
+                validation_type=self.has_no_ports_when_required.__name__,
+            )
+
+    def has_edgeport_when_forbidden(self, simulation, solution):
+        """Validation check: ensure that if at least one "EdgePort" is present, some solution types can't be chosen.
+        Args:
+            simulations: A Simulation object.
+            solutions: A Solution object.
+        Raises:
+            Errors when validation criteria are not met.
+        """
+        port_names = get_port_names(simulation)
+        sim_name = simulation.name
+        if "EdgePort" in port_names and type(solution) in [
+            AnsysEigenmodeSolution,
+            AnsysVoltageSolution,
+            AnsysCurrentSolution,
+        ]:
+            raise ValidateSimError(
+                f"Simulation '{sim_name}' has at least one 'EdgePort'. This is incompatible with {type(solution)}",
+                validation_type=self.has_edgeport_when_forbidden.__name__,
+            )
+
+    def flux_integration_layer_exists_if_needed(self, simulation, solution):
+        """Validation check related to the presence of layers and magnetic flux integration.
+        Args:
+            simulation: A Simulation object.
+        Raises:
+            Errors when validation criteria are not met.
+        """
+        sim_name = simulation.name
+        has_integrate_flux = hasattr(solution, "integrate_magnetic_flux")
+        integrate_flux = solution.integrate_magnetic_flux if has_integrate_flux else False
+
+        # Ensures that a layer with thickness == 0 and a material != "pec"
+        # exists in the setup when "integrate_magnetic_flux" is True.
+        if integrate_flux:
+            layers = simulation.layers
+            has_flux_integration_layer = False
+            for layer in layers.values():
+                if layer.get("thickness", -1) == 0 and layer.get("material", "") != "pec":
+                    has_flux_integration_layer = True
+                    break
+            if not has_flux_integration_layer:
+                raise ValidateSimError(
+                    f"Simulation '{sim_name}' has 'integrate_magnetic_flux = True' 
+                    but the integration layer is missing.",
```

Comment:
```suggestion
                    f"Simulation '{sim_name}' has 'integrate_magnetic_flux = True' "
                    + "but the integration layer is missing.",
```
This still fails documentation building for some reason, maybe change it to this?

Reply:
This actually fails many things :) But fixing it as I suggested should work